File: programmers/lv1/약수의개수와덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        nums = []
        for j in range(1, int(i**(1/2))+1):
            if i % j == 0:
                nums.append(i // j)
                if j * j != i:
                    nums.append(j)
        if len(nums) % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer
solution(13, 17)

File: programmers/lv1/약수의개수와덧셈.js
const gcdList = (num) => {
  let list = [];
  for (let i = 1; i <= parseInt(num ** (1 / 2)); i++) {
    if (num % i == 0) {
      let n = parseInt(num / i);
      list.push(i);
      if (i != n) list.push(n);
    }
  }
  return list;
};

function solution(left, right) {
  var answer = 0;

  for (let i = left; i <= right; i++) {
    const list = gcdList(i);
    if (list.length % 2 == 0) answer += i;
    else answer -= i;
  }
  return answer;
}
